From 1ce6b40da8ecbf997c5522a592a122e805aa81c4 Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@igalia.com>
Date: Wed, 3 Apr 2019 18:00:55 -0500
Subject: [PATCH] gnutls: Handle new GNUTLS_E_CERTIFICATE_REQUIRED

This new error in GnuTLS 3.6.7 should correspond to
G_TLS_ERROR_CERTIFICATE_REQUIRED. This fixes our client auth tests when
run with GnuTLS 3.6.7.

Fixes #70
---
 tls/gnutls/gtlsconnection-gnutls.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tls/gnutls/gtlsconnection-gnutls.c b/tls/gnutls/gtlsconnection-gnutls.c
index 27bbbcd..774d668 100644
--- a/tls/gnutls/gtlsconnection-gnutls.c
+++ b/tls/gnutls/gtlsconnection-gnutls.c
@@ -1062,7 +1062,11 @@ end_gnutls_io (GTlsConnectionGnutls  *gnutls,
       else
         return 0;
     }
-  else if (status == GNUTLS_E_NO_CERTIFICATE_FOUND)
+  else if (status == GNUTLS_E_NO_CERTIFICATE_FOUND
+#ifdef GNUTLS_E_CERTIFICATE_REQUIRED
+           || status == GNUTLS_E_CERTIFICATE_REQUIRED /* Added in GnuTLS 3.6.7 */
+#endif
+          )
     {
       g_set_error_literal (error, G_TLS_ERROR, G_TLS_ERROR_CERTIFICATE_REQUIRED,
                            _("TLS connection peer did not send a certificate"));
-- 
2.18.1

